File: Assignments/Assignment1/TranDinhDuyPhuong_SE18D04/src/services/tagService.js
import api from './api';

const getAllTags = async () => {
    const response = await api.get('/tags');
    return response.data;
};

const getTagById = async (id) => {
    const response = await api.get(`/tags/${id}`);
    return response.data;
};

const createTag = async (tagData) => {
    const response = await api.post('/tags', tagData);
    return response.data;
};

const updateTag = async (id, tagData) => {
    const response = await api.put(`/tags/${id}`, tagData);
    return response.data;
};

const deleteTag = async (id) => {
    const response = await api.delete(`/tags/${id}`);
    return response.data;
};

const searchTags = async (keyword) => {
    const response = await api.get(`/tags/search?keyword=${keyword}`);
    return response.data;
};

const tagService = {
    getAllTags,
    getTagById,
    createTag,
    updateTag,
    deleteTag,
    searchTags
};

export default tagService;

File: Assignments/Assignment2/A2TranDinhDuyPhuong_SE18D04/src/main/java/sba301/a2trandinhduyphuong_se18d04/services/TagService.java
package sba301.a2trandinhduyphuong_se18d04.services;

import sba301.a2trandinhduyphuong_se18d04.dtos.request.TagRequest;
import sba301.a2trandinhduyphuong_se18d04.dtos.response.TagResponse;

import java.util.List;

public interface TagService {

    List<TagResponse> getAllTags();

    TagResponse getTagById(Long id);

    TagResponse createTag(TagRequest request);

    TagResponse updateTag(Long id, TagRequest request);

    void deleteTag(Long id);

    List<TagResponse> searchTags(String keyword);

}
